pkg/api/types.go

```diff
@@ -1662,6 +1665,27 @@ type SecretList struct {
 	Items []Secret `json:"items"`
 }
 
+// SecurityContextSpec holds security configuration that will be applied to the pod's containers
+type SecurityContextSpec struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+
+	// Capabilities represents the caps to add or remove from all containers in the pod
+	Capabilities Capabilities `json:"capabilities,omitempty"`
+
```

Comment:
Containers should run as a user.  The context is declarative:

* I want to run as user ID 6
* I want to run as group ID 10
* I want to run privileged
* I want to add these capabilities
* I want to drop these capabilities
* I want to run with user namespaces in range mode using the UIDs from 2,810,000 to 2,820,000.

The security provider in the kubelet, and the security context that is associated with the service account, will say what the containers can do:

* Containers under this security context can:
  * ask to be run as privileged
  * run as only this user and god
  * run as any user except root
  * must run as a random UID on the host
  * may ask for these capabilities